cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/ZookeeperSetOperator.java

```diff
@@ -54,4 +102,225 @@ public static boolean needsRollingUpdate(StatefulSetDiff diff) {
+            X509Certificate caCertCO = (X509Certificate) x509.generateCertificate(new ByteArrayInputStream(decoder.decode(clusterCaCertificateSecret.getData().get("ca.crt"))));
+
+            KeyManagerFactory kmf = KeyManagerFactory.getInstance("SunX509");
+            kmf.init(setupKeyStore(clusterCaKeySecret, x509, password, caCertCO, coCertKey), password);
+
+            TrustManagerFactory tmf = TrustManagerFactory.getInstance("X509");
+            tmf.init(setupTrustStore(password, caCertCO));
+            SSLContext ctx = SSLContext.getInstance("TLS");
+            ctx.init(kmf.getKeyManagers(), tmf.getTrustManagers(), null);
+            factory = ctx.getSocketFactory();
+        } catch (CertificateException | NoSuchAlgorithmException | UnrecoverableKeyException | KeyStoreException | KeyManagementException e) {
+            log.error(e.getMessage());
+        }
+        return factory;
+    }
+
+    private boolean isLeader(Pod pod, SSLSocketFactory factory) {
+        int port = 2181;
+        boolean leader = false;
+        try {
+            String host = pod.getStatus().getPodIP();
+
+            SSLSocket socket = null;
+            Socket plainSocket = factory.createSocket();
+            if (plainSocket instanceof SSLSocket) {
+                // findbugs plugin does not like direct overtyping
+                socket = (SSLSocket) plainSocket;
+            }
+            log.debug("Connecting client to {}:{}", host, port);
+            try {
+                socket.connect(new InetSocketAddress(host, port), 10_000);
+            } catch (ConnectException | SocketTimeoutException e) {
+                log.error("Could not connect " + e.getMessage());
+            }
+            try {
+                log.debug("Starting handshake with {}", socket.getRemoteSocketAddress());
+                try {
+                    socket.startHandshake();
+                    PrintWriter out = new PrintWriter(
+                            new BufferedWriter(
+                                    new OutputStreamWriter(
+                                            socket.getOutputStream(), StandardCharsets.UTF_8)));
+
+                    out.println("stat");
+                    out.flush();
+
+                    BufferedReader in = new BufferedReader(
+                            new InputStreamReader(
+                                    socket.getInputStream(), StandardCharsets.UTF_8));
+
+                    String inputLine;
+                    while ((inputLine = in.readLine()) != null) {
+                        log.debug(inputLine);
+                        if (inputLine.equals("Mode: leader")) {
+                            leader = true;
+                        }
+                    }
+                    in.close();
+                    out.close();
+                } catch (SSLHandshakeException e) {
+                    log.error("handshake exception - " + e.getMessage());
+                    e.printStackTrace();
```

Comment:
Replace these two lines with `log.error("Error while performing TLS handshake with pod {} in namespace {}", pod.getMetadata().getName(), pod.getMetatate().getNamespace(), e);`